test/ILLink.RoslynAnalyzer.Tests/DynamicallyAccessedMembersAnalyzerTests.cs

```diff
@@ -1018,5 +1018,82 @@ private static void M2<S>()
 				.WithSpan (17, 9, 17, 16)
 				.WithArguments ("T", "C.M1<T>()", "S", "C.M2<S>()", "'DynamicallyAccessedMemberTypes.PublicMethods'"));
 		}
+
+		[Fact]
+		public Task SourceTypeofFlowsIntoTargetMethodReturnTypeAnnotation ()
+		{
+			var TargetMethodReturnTypeWithAnnotations = @"
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+public class T
+{
+}
+
+class C
+{
+    public static void Main()
+    {
+        M();
+    }
+
+    [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+    private static Type M()
+    {
+        return typeof(C);
+    }
+}";
+
+			// TODO: https://github.com/dotnet/linker/issues/2308
+			// This should produce no warnings. Currently it produces the following:
```

Comment:
Should we have this test verify the expected result (i.e., no warnings produced) and use `[Fact(Skip = "This should...")]` instead?

Reply:
I was going back and for on this - done!